simulator/src/main/java/com/hedera/block/simulator/exception/BlockSimulatorParsingException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
What's the advantage of having our own checked exception rather than using ParseException?

Reply:
So the main reason of me adding this is because I wanna limit our dependencies on external libraries, otherwise we would have to constantly introduce new libraries in modules that use simulator, in this case suites. In this case ParseException is coming from a external source. Another side effect and advantage of this can be better control. Now if, we want to do something or handle in specific way, we could do that.
Although, if you think that it's unnecessary and pointless, I will remove 👍